test/kt/single_pass_scan.cpp

```diff
@@ -0,0 +1,147 @@
+        sycl::buffer<T> buf_out(input.size());
+        oneapi::dpl::experimental::kt::inclusive_scan(q, view, view, ::std::plus<T>(),
+                                                                      param)
+            .wait();
+    }
+
+    std::string msg = "wrong results with all_view, n: " + std::to_string(size);
+    EXPECT_EQ_RANGES(ref, input, msg.c_str());
+}
+#endif
+
+template <typename T, sycl::usm::alloc _alloc_type, typename KernelParam>
+void
+test_usm(sycl::queue q, std::size_t size, KernelParam param)
+{
+#if LOG_TEST_INFO
+    std::cout << "\t\ttest_usm<" << TypeInfo().name<T>() << ", " << USMAllocPresentation().name<_alloc_type>()
+              ">(" << size << ");" << std::endl;
+#endif
+    std::vector<T> expected(size);
+    generate_data(expected.data(), size, 42);
+
+    TestUtils::usm_data_transfer<_alloc_type, T> dt_input(q, expected.begin(), expected.end());
+    TestUtils::usm_data_transfer<_alloc_type, T> dt_output(q, size);
+
+    std::inclusive_scan(expected.begin(), expected.end(), expected.begin());
+
+    oneapi::dpl::experimental::kt::inclusive_scan(q, dt_input.get_data(), dt_input.get_data() + size, dt_output.get_data(), ::std::plus<T>(), param)
+        .wait();
+
+    std::vector<T> actual(size);
+    dt_output.retrieve_data(actual.begin());
+
+    std::string msg = "wrong results with USM, n: " + std::to_string(size);
+    EXPECT_EQ_N(expected.begin(), actual.begin(), size, msg.c_str());
+}
+
+template <typename T, typename KernelParam>
+void
+test_sycl_iterators(sycl::queue q, std::size_t size, KernelParam param)
+{
+#if LOG_TEST_INFO
+    std::cout << "\t\ttest_sycl_iterators<" << TypeInfo().name<T>() << ">(" << size << ");" << std::endl;
+#endif
+    std::vector<T> input(size);
+    generate_data(input.data(), size, 42);
+    std::vector<T> ref(input);
+    std::inclusive_scan(std::begin(ref), std::end(ref), std::begin(ref));
+    {
+        sycl::buffer<T> buf(input.data(), input.size());
+        sycl::buffer<T> buf_out(input.size());
+        oneapi::dpl::experimental::kt::inclusive_scan(q, oneapi::dpl::begin(buf), oneapi::dpl::end(buf), oneapi::dpl::begin(buf_out), ::std::plus<T>(),
+                                                                      param)
+            .wait();
+    }
+
+    std::string msg = "wrong results with oneapi::dpl::begin/end, n: " + std::to_string(size);
+    EXPECT_EQ_RANGES(ref, input, msg.c_str());
+}
+
+template <typename T, typename KernelParam>
+void
+test_general_cases(sycl::queue q, std::size_t size, KernelParam param)
+{
+    test_usm<T, sycl::usm::alloc::shared>(q, size, param);
+    test_usm<T, sycl::usm::alloc::device>(q, size, param);
+    test_sycl_iterators<T>(q, size, param);
+#if _ENABLE_RANGES_TESTING
+    test_all_view<T>(q, size, param);
+#endif
+}
+
+
+int
+main()
+{
+    using T = int;
+    constexpr int data_per_work_item = 8;
+    constexpr int work_group_size = 256;
```

Comment:
Because we already have some technique to generate the set of tests in the `kt` test folder, I propose to use this technique to generate one test for one set of parameters values too.
For details please see `CMakeLists.txt` in this folder.

Reply:
Done.